creator-node/src/routes/files.js

```diff
@@ -192,13 +192,34 @@ module.exports = function (app) {
       if (req.query.filename) {
         res.setHeader('Content-Disposition', contentDisposition(req.query.filename))
       }
+
+      // Determine if cat one byte of ipfs CID takes over 500ms
+      await ipfsSingleByteCat(CID, req, 500)
+
+      // Stream file from ipfs if cat one byte takes under 500ms
       await new Promise((resolve, reject) => {
         req.app.get('ipfsAPI').catReadableStream(CID)
           .on('data', streamData => { res.write(streamData) })
           .on('end', () => { res.end(); resolve() })
           .on('error', e => { reject(e) })
       })
     } catch (e) {
+      // ipfsCatSingleByte took over 500ms, try streaming from file system
```

Comment:
this comment seems incomplete, as this codepath can be executed either if ipfsSingleByteCat() throws (ie exceeds 500ms), or if the catReadableStream() promise rejects. can you update accordingly?